src/admin/js/alma-admin-modal-checkout-legal.js

```diff
@@ -27,8 +27,24 @@ jQuery(document).ready(function () {
             'accept': value
         };
 
-        jQuery.post(ajax_object.ajax_url, data, function() {
-            location.reload();
+        var modalSoc = jQuery('#alma-modal-soc');
+
+        jQuery.post(ajax_object.ajax_url, data, function(response) {
```

Comment:
the anonymous function given as `jQuery.post` third parameter is only called on success according to [this documentation](https://api.jquery.com/jquery.post/#jQuery-post-url-data-success-dataType)

Please use `.done(function() {})` and `.fail(function() {})` to handle success and error instead of using the default success third argument.

Reply:
Fixed